api4/handlers_test.go

```diff
@@ -0,0 +1,86 @@
+	io.WriteString(w, body)
+}
+
+func TestApiHandlerWithGzip(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	WebserverMode := *th.App.Config().ServiceSettings.WebserverMode
+
+	defer func() {
+		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.WebserverMode = WebserverMode })
+	}()
+
+	// WebserverMode = "gzip"
+	th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.WebserverMode = "gzip" })
+
+	api := Init(th.Server, th.Server.AppOptions, th.Server.Router)
+
+	handler := api.ApiHandler(handlerForGzip)
+
+	// No specified "Accept-Encoding" in req
```

Comment:
Instead of just using comments to mark sections of the tests, we've been switching more tests to use `t.Run(name string, func(t *testing.T))` to improve readability like
```go
th := Setup().InitBasic()
defer th.TearDown()

api := Init(th.Server, th.Server.AppOptions, th.Server.Router)

t.Run("with WebServerMode set to gzip", func(t *testing.T) {
    th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.WebserverMode = "gzip" })

    handler := api.ApiHandler(handlerForGzip)

    t.Run("with Accept-Encoding set to gzip", func(t *Testing.T) { /* ... */})

    t.Run("without Accept-Encoding", func(t *Testing.T) { /* ... */})
})

t.Run("with WebServerMode set to uncompressed", func(t *testing.T) {
    th.App.UpdateConfig(func(cfg *model.Config) { *cfg.ServiceSettings.WebserverMode = "uncompressed" })

    handler := api.ApiHandler(handlerForGzip)

    // ...
})
```